### DISCUSSION TITLE
Utility of ChampSim

### DISCUSSION THREAD

**Discussion by Pavitra07** [ChampSim/ChampSim#370] [Q&A]
I am working on coming up with a custom cache structure and replacement policy. Basically the structure of cache would have additional bits to each cache line and the replacement would involve address encryptions too. Will Champsim simulator be useful to conduct such a task?

**Comment by ngober** [comment]
I think you could probably do this. One consideration you will need is that your replacement policy must keep its own metadata. There is not a place to add bits to a cache line, but a replacement policy could keep a separate structure that contains it.

**Reply by Pavitra07** [reply]
I was actually interested in adding more bits to each cache line to store information pertaining to the address mapped. If that is not possible in ChampSim, could you please guide some other tool or simulator that would be helpful to do so?

**Reply by ngober** [reply]
As I said, the preferred way to do this would be to maintain your own metadata structure. If you modify the cache block structure, you will need to dig into the internals of the simulator. I anticipate this would apply to any tool or simulator you ultimately choose.
